### PR TITLE
[v2][adjuster] Implement function to get standard adjusters to operate on otlp format

### PR DESCRIPTION
**Pull request by mahadzaryab1**
## Which problem is this PR solving?
- Towards #6344 

## Description of the changes
- Implemented a function `StandardAdjusters` that returns a list of adjusters to be applied on ptrace.Traces
- This will be used by the v2 query service in https://github.com/jaegertracing/jaeger/pull/6343

## Checklist
- [x] I have read https://github.com/jaegertracing/jaeger/blob/master/CONTRIBUTING_GUIDELINES.md
- [x] I have signed all commits
- [x] I have added unit tests for the new functionality
- [x] I have run lint and test steps successfully
  - for `jaeger`: `make lint test`
  - for `jaeger-ui`: `npm run lint` and `npm run test`

### CODE REVIEW

File: cmd/query/app/querysvc/adjuster/hash.go [hunk 1]
```diff
@@ -33,7 +33,7 @@ type SpanHashDeduper struct {
 	marshaler ptrace.Marshaler
 }
 
-func (s *SpanHashDeduper) Adjust(traces ptrace.Traces) {
+func (s SpanHashDeduper) Adjust(traces ptrace.Traces) {
```
[the remark this answers is not in the document]
Comment (mahadzaryab1) [reply cmd/query/app/querysvc/adjuster/hash.go:36]: sorry thought i reverted this change - let me fix

File: cmd/query/app/querysvc/adjuster/standard.go [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+// before returning the data to the API clients.
+func StandardAdjusters(maxClockSkewAdjust time.Duration) []Adjuster {
+	return []Adjuster{
+		SpanIDUniquifier(),
+		SortAttributesAndEvents(),
+		SpanHash(),                    // requires SortAttributesAndEvents for deterministic results
+		ClockSkew(maxClockSkewAdjust), // adds warnings (which affect SpanHash) on duplicate span IDs
+		IPAttribute(),
+		ResourceAttributes(),
+		SpanLinks(),
```
Comment (yurishkuro) [inline cmd/query/app/querysvc/adjuster/standard.go:20]: We should consider renaming these adjusters to reflect the business function. Reading this list I can reasonably guess what the first two are doing, plus ClockSkew. But all others are poorly named.

My suggestion would be to use verbs that describe the function, e.g.
  * DedupeClientServerSpanIDs
  * SortAttributes
  * DedupeSpans
  * etc

Comment (mahadzaryab1) [reply cmd/query/app/querysvc/adjuster/standard.go]: @yurishkuro Done. Let me know what you think

File: cmd/query/app/querysvc/adjuster/standard.go [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+	"time"
+)
+
+// StandardAdjusters returns a list of adjusters applied by the query service
+// before returning the data to the API clients.
+func StandardAdjusters(maxClockSkewAdjust time.Duration) []Adjuster {
+	return []Adjuster{
+		DeduplicateClientServerSpanIDs(),
+		SortCollections(),
+		DeduplicateSpans(),                   // requires SortAttributesAndEvents for deterministic results
```
Comment (yurishkuro) [inline cmd/query/app/querysvc/adjuster/standard.go:16]:
```suggestion
		// DeduplicateSpans depends on SortCollections running first
		DeduplicateSpans(),
```

File: cmd/query/app/querysvc/adjuster/standard.go [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+		CorrectClockSkew(maxClockSkewAdjust), // adds warnings (which affect SpanHash) on duplicate span IDs
```
Comment (yurishkuro) [inline cmd/query/app/querysvc/adjuster/standard.go:17]: is the comment helpful / relevant here?

**Review by yurishkuro — APPROVED** [review]

File: cmd/query/app/querysvc/adjuster/standard.go (hunk 4 above)
Comment (mahadzaryab1) [reply cmd/query/app/querysvc/adjuster/standard.go]: removed

**Review by yurishkuro — APPROVED** [review]

**Pull request merged**